[fix #7748] Add lifecycle/welcome page 2

craigcook opened the pull request:
## Description
Adds a new lifecycle page promoting Pocket.

## Issue / Bugzilla link
#7748 

## Testing
https://bedrock-demo-craigcook.oregon-b.moz.works/firefox/welcome/2/

amychurchwell commented on media/js/firefox/welcome-page2.js:
```diff
@@ -0,0 +1,68 @@
+
+        var destURL = pocketButton.getAttribute('data-action') + 'metrics-flow';
+
+        // collect values from monitor button
```
copypasta- change `monitor button` to `pocket button` 🍝 :-)

amychurchwell commented on bedrock/firefox/templates/firefox/welcome/page2.html:
```diff
@@ -0,0 +1,65 @@
+{% endblock %}
+
+{% block content_utility %}
+  <p><strong><a href="https://support.mozilla.org/kb/firefox-browser-welcome-pages/?utm_source=mozilla.org-firefox-welcome-2&utm_medium=referral&utm_campaign=welcome-2-pocket&entrypoint=mozilla.org-firefox-welcome-2">{{ _('Why am I seeing this?') }}</a></strong></p>
```
not a blocker per se; but I think we should make utm params variables at the top of templates. imo, it is a good way to ensure that params are consistent and maintainable. 
For example:
```
{% set _entrypoint = 'mozilla.org-firefox-welcome-2' %}
{% set _utm_campaign = 'welcome-2-pocket' %}
. . . 
<a href="https://support.mozilla.org/kb/firefox-browser-welcome-pages/?utm_source={{ _entrypoint }}&utm_medium=referral&utm_campaign={{ _utm_campaign }}&entrypoint={{ _entrypoint }}">
```

amychurchwell commented on bedrock/firefox/templates/firefox/welcome/page2.html:
```diff
@@ -0,0 +1,65 @@
+{% block content_intro %}
+  <h1 class="c-page-title">{{ _('Your time online is worth protecting') }}</h1>
+  {# L10n: "Pocket" is a product name and shouldn't be translated. #}
+  <p class="c-page-tagline">{{ _('Discover and save stories in Pocket — and come back to them when you’re free.') }}</p>
```
- I think that we should consider making `page-intro` into a `mzp-c-hero`.
<img width="1254" alt="Screen Shot 2019-10-15 at 12 34 50 PM" src="https://user-images.githubusercontent.com/30483988/66863944-f34b6880-ef48-11e9-80db-727ec10d2b8f.png">

- I also think we should strongly consider adding the "Activate Pocket" button to the hero section. Especially considering that this page is very similar in function to the whatsnew page.

- We should also write some tests for the new pocket button. :+1:

craigcook replied:
I tried changing it to the hero but then I had to customize it because the design doesn't use the standard component (different sizes, padding, etc). So I'm not sure which is worse: a customized component that can break if the dependency changes upstream, or another non-standard, non-component. Feels like debt either way.

The button isn't part of the hero because the design didn't put it there. This one is following the pattern established on the first page so if we move the button we should do it on both. I'm reluctant to make that kind of design change at this stage though since these have already been through so much review. Seems like a good thing we could test later though.

I'm working on adding a test for the button at least.

amychurchwell replied:
Thanks Craig! That's all fair enough. Considering that this is a set of pages, in the future we should push to make these templates modular in protocol like the whatsnew page. (is preaching to choir, haha)

amychurchwell commented on bedrock/firefox/templates/firefox/welcome/page2.html:
```diff
@@ -0,0 +1,65 @@
+    <img class="primary-image" src="{{ static('img/firefox/welcome/pocket-hero.svg') }}" width="700" height="" alt="">
+
+    <p class="primary-cta">
+      <a id="pocket-cta" class="mzp-c-button mzp-t-product" href="https://getpocket.com/ff_signup?s=ffwelcome2&form_type=button&entrypoint=mozilla.org-firefox-welcome-2&utm_source=mozilla.org-firefox-welcome-2&utm_campaign=welcome-2-pocket&utm_medium=referral" data-action="{{ settings.FXA_ENDPOINT }}" data-cta-text="Activate Pocket" data-cta-type="FxA-Pocket" data-cta-position="Primary" rel="exte ...
```
Is it possible for folks who do not have a firefox account to be shown this page? Afaik, I think this page is only shown to fxa users, but just wanted to double-check.

craigcook replied:
I believe it's shown to people X days after they create a new profile, which is usually a new user so many if not most of them will not have a Firefox account. Promoting other products is an incentive to join.

craigcook commented on bedrock/firefox/templates/firefox/welcome/page2.html:
```diff
@@ -0,0 +1,65 @@
+
+{% block page_image %}{{ static('img/firefox/template/page-image-master.jpg') }}{% endblock %}
+
+{% block body_class %}welcome-page2{% endblock %}
```
We're not actually using it for anything at this point, I originally included it just in case we needed some style hook to differentiate pages.

stephaniehobson replied:
Don't you need it for the heading fonts to be in Metropolis?

craigcook replied:
Oh I see... yeah this needs a `super()`, sorry I missed that.

craigcook commented:
Ready for re-review. I did end up changing to the hero component. I also added a page image for social sharing, which I added to the Pocket page as well (Stephanie requested it).
